jedi/evaluate/docstrings.py

```diff
@@ -45,28 +49,147 @@
+        returns += doc._parsed_data['Yields']
+        found = []
+        for p_name, p_type, p_descr in returns:
+            if not p_type:
+                p_type = p_name
+                p_name = ''
+
+            m = re.match('([^,]+(,[^,]+)*?)$', p_type)
+            if m:
+                p_type = m.group(1)
+            found.extend(_expand_typestr(p_type))
+        return found
+
+
+def _expand_typestr(p_type):
+    """
+    Attempts to interpret the possible types
+    """
+    # Check if alternative types are specified
+    if re.search('\\bor\\b', p_type):
+        types = [t.strip() for t in p_type.split('or')]
+    # Check if type has a set of valid literal values
+    elif p_type.startswith('{'):
+        if not is_py3:
+            # python2 does not support literal set evals
+            # workaround this by using lists instead
+            p_type = p_type.replace('{', '[').replace('}', ']')
+        types = set(type(x).__name__ for x in literal_eval(p_type))
+        types = list(types)
+    # Otherwise just return the typestr wrapped in a list
+    else:
+        types = [p_type]
+    return types
 
-def _search_param_in_docstr(docstr, param_str):
+
+def _search_param_in_googledocstr(docstr, param_str):
+    r"""
+    >>> from jedi.evaluate.docstrings import *  # NOQA
+    >>> from jedi.evaluate.docstrings import _search_param_in_googledocstr
+    >>> docstr = (
+    ... 'Args:\n'
+    ... '    x (ndarray):\n'
+    ... '    y (int or str or list):\n'
+    ... '    z ({"foo", "bar", 100500}):\n'
+    ... )
+    >>> _search_param_in_googledocstr(docstr, 'x')
+    ['ndarray']
+    >>> sorted(_search_param_in_googledocstr(docstr, 'y'))
+    ['int', 'list', 'str']
+    >>> sorted(_search_param_in_googledocstr(docstr, 'z'))
+    ['int', 'str']
+    """
+    found = None
+    for garg in docscrape_google.parse_google_args(docstr):
+        if garg['name'] == param_str:
+            typestr = garg['type']
+            found = _expand_typestr(typestr)
+            break
+    return found
+
+
+def _search_return_in_gooogledocstr(docstr):
+    r"""
+    >>> from jedi.evaluate.docstrings import *  # NOQA
+    >>> from jedi.evaluate.docstrings import _search_return_in_gooogledocstr
+    >>> docstr = (
+    ... 'Returns:\n'
+    ... '    ndarray:\n'
+    ... '    int\n'
+    ... )
+    >>> sorted(_search_return_in_gooogledocstr(docstr))
+    ['int', 'ndarray']
     """
+    google_rets = list(docscrape_google.parse_google_returns(docstr))
+    found = []
+    for retdict in google_rets:
+        p_type = retdict['type']
+        found.extend(_expand_typestr(p_type))
+    return list(set(found))
+
+
+def _search_param_in_docstr(docstr, param_str):
+    r"""
     Search `docstr` for type(-s) of `param_str`.
 
+    >>> from jedi.evaluate.docstrings import _search_param_in_docstr
+    >>> from jedi.evaluate.docstrings import _search_param_in_googledocstr
+    >>> from jedi.evaluate.docstrings import _search_param_in_numpydocstr
```

Comment:
I don't think you need these imports here? You can just omit them, because we're in the same module, can't you? I might be wrong as well.